bin/completed_segments.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+""" Find (and delete) fully completed kafka-backup segments """
+
+import os
+import re
+import sys
+import argparse
+
+
+def filename_pattern():
+    """ Kafka-backup segment/index filename pattern """
+    return 'segment_partition_%s_from_offset_%s_%s'
+
+
+def parse_args():
+    """ Parse cmdline args """
+    parser = argparse.ArgumentParser(
+        description='Find (and delete) fully completed kafka-backup segments'
+    )
+    parser.add_argument(
+        '-d', '--delete',
+        help='Delete fully completed segment files',
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        'target_dir',
+        help='Backup directory (target.dir)',
+        default=os.getcwd(),
+    )
+    return parser.parse_args()
+
+
+def collect_segments(target_dir):
```

Comment:
Just thinking whats the cleanest approach: using the partition index to identify the segments or traversing the files…
Even thinking in #87  whether to drop the index files altogether as they are "just" a performance optimiziation. Any ideas?

Reply:
Files traversal looks easier from my POV as I don't need to parse partition files (don't need to even know about what's there).
I have no opinion on index files so far. Just doing things consistently (remove *_records => remove *_index too). So here it's up to you and restoration tool/script/guide :)

Reply:
Then lets keep the file traversal.